### ISSUE TITLE
Ruby version 3.2.0

### ISSUE THREAD

**Issue by elalemanyo** [sds/scss-lint#1000]
Hi,
I am trying to upgrade to ruby version 3.2.0, unfortunately there seems to be some error:
```ruby
SCSSLint::Linter::ChainedClasses raised unexpected error linting file app/assets/stylesheets/admin/base/_base.scss: 'undefined method `=~' for a:Sass::Selector::SimpleSequence'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/linter/chained_classes.rb:9:in `block in visit_sequence'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/linter/chained_classes.rb:8:in `each'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/linter/chained_classes.rb:8:in `visit_sequence'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/selector_visitor.rb:12:in `visit_selector_node'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/selector_visitor.rb:5:in `visit_selector'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/selector_visitor.rb:21:in `block in visit_members'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/selector_visitor.rb:20:in `each'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/selector_visitor.rb:20:in `visit_members'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/selector_visitor.rb:14:in `visit_selector_node'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/selector_visitor.rb:5:in `visit_selector'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/linter.rb:139:in `visit'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/linter.rb:154:in `block in visit_children'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/linter.rb:152:in `each'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/linter.rb:152:in `visit_children'
/usr/local/bundle/gems/sass-3.7.4/lib/sass/tree/visitors/base.rb:38:in `visit'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/linter.rb:143:in `visit'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/linter.rb:41:in `run'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/runner.rb:53:in `run_linter'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/runner.rb:35:in `block in find_lints'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/runner.rb:33:in `each'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/runner.rb:33:in `find_lints'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/runner.rb:19:in `block in run'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/runner.rb:18:in `each'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/runner.rb:18:in `run'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/cli.rb:67:in `scan_for_lints'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/cli.rb:59:in `act_on_options'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/lib/scss_lint/cli.rb:33:in `run'
/usr/local/bundle/bundler/gems/scss-lint-7b74cdfb86c0/bin/scss-lint:7:in `<top (required)>'
/usr/local/bundle/bin/scss-lint:25:in `load'
/usr/local/bundle/bin/scss-lint:25:in `<top (required)>'
/usr/local/bundle/gems/bundler-2.3.8/lib/bundler/cli/exec.rb:58:in `load'
/usr/local/bundle/gems/bundler-2.3.8/lib/bundler/cli/exec.rb:58:in `kernel_load'
/usr/local/bundle/gems/bundler-2.3.8/lib/bundler/cli/exec.rb:23:in `run'
/usr/local/bundle/gems/bundler-2.3.8/lib/bundler/cli.rb:484:in `exec'
/usr/local/bundle/gems/bundler-2.3.8/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/usr/local/bundle/gems/bundler-2.3.8/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/usr/local/bundle/gems/bundler-2.3.8/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
/usr/local/bundle/gems/bundler-2.3.8/lib/bundler/cli.rb:31:in `dispatch'
/usr/local/bundle/gems/bundler-2.3.8/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
/usr/local/bundle/gems/bundler-2.3.8/lib/bundler/cli.rb:25:in `start'
/usr/local/bundle/gems/bundler-2.3.8/exe/bundle:48:in `block in <top (required)>'
/usr/local/bundle/gems/bundler-2.3.8/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
/usr/local/bundle/gems/bundler-2.3.8/exe/bundle:36:in `<top (required)>'
/usr/local/bundle/bin/bundle:25:in `load'
/usr/local/bundle/bin/bundle:25:in `<main>'
```
It is not the file itself, it seems to be a more general problem.

- SCSS-Lint version: 0.59.0
- Sass version: 3.7.4
- Ruby version: 3.2.0
- .scss-lint.yml
```yaml
linters:
  BangFormat:
    space_before_bang: true
    space_after_bang: false
  BemDepth:
    max_elements: 1
  BorderZero:
    convention: zero
  PropertySortOrder:
    order: concentric
  ColorVariable:
    enabled: false
  ChainedClasses:
    enabled: true
  EmptyLineBetweenBlocks:
    enabled: false
  EmptyRule:
    enabled: true
  FinalNewline:
    enabled: true
  HexLength:
    style: short
  HexValidation:
    enabled: true
  IdSelector:
    enabled: true
  SelectorFormat:
    convention: hyphenated_BEM
  SelectorDepth:
    max_depth: 3
  ImportantRule:
    enabled: false
  LeadingZero:
    enabled: false
  PseudoElement:
    enabled: true
  Comment:
    enabled: false
  Indentation:
    severity: warning
    width: 2
  MergeableSelector:
    enabled: false
  NestingDepth:
    max_depth: 3
  Shorthand:
    allowed_shorthands: [1, 2, 3, 4]
  SingleLinePerProperty:
    allow_single_line_rule_sets: false
  SpaceAfterComma:
    style: one_space
  SpaceAfterPropertyColon:
    style: one_space
  SpaceAfterPropertyName:
    style: no_space
  SpaceAfterVariableColon:
    style: one_space
  SpaceAfterVariableName:
    style: no_space
  SpaceAroundOperator:
    style: one_space
  SpaceBeforeBrace:
    style: one_space
  StringQuotes:
    style: single_quotes
  VariableForProperty:
    enabled: false
  UnnecessaryParentReference:
    enabled: false
  ZeroUnit:
    enabled: true
```

Thank you for any help

**Comment by sds** [comment]
I don't have an immediate answer to your question. Would welcome any help digging into the underlying cause.

**Comment by elalemanyo** [comment]
It seems that the problem is in [`chained_classes.rb#L9`](https://github.com/sds/scss-lint/blob/master/lib/scss_lint/linter/chained_classes.rb#L9). Running the tests using ruby 3.2.0 I get the same error:

```
SCSSLint::Linter::ChainedClasses with a chained class in a nested rule set
Failure/Error: line_offset += 1 if member =~ /\n/
NoMethodError:
  undefined method `=~' for p:Sass::Selector::SimpleSequence
```

I haven't made much progress, but I wanted to document it.

**Comment by elalemanyo** [comment]
@sds closing here. Thanks for all your help 🙂
